packages/math/is_prime.pony

```diff
@@ -0,0 +1,80 @@
+primitive IsPrime[A: (Integer[A] val & Unsigned)]
+  """Primality test using 6k+-1 optimization."""
+
+  // Using a match table like below provides information to LLVM
+  // to allocate static globals rather than stack or heap allocation.
+  fun _low_prime_table_size(): USize => 256
+  fun _low_prime_table(index: USize): A val =>
+    match index
+    | 0 => 2 | 1 => 3 | 2 => 5 | 3 => 7 | 4 => 11 
+    | 5 => 13 | 6 => 17 | 7 => 19 | 8 => 23 | 9 => 29 
+    | 10 => 31 | 11 => 37 | 12 => 41 | 13 => 43 | 14 => 47 
+    | 15 => 53 | 16 => 59 | 17 => 61 | 18 => 67 | 19 => 71 
+    | 20 => 73 | 21 => 79 | 22 => 83 | 23 => 89 | 24 => 97
+    | 25 => 101 | 26 => 103 | 27 => 107 | 28 => 109 | 29 => 113
+    | 30 => 127 | 31 => 131 | 32 => 137 | 33 => 139 | 34 => 149
+    | 35 => 151 | 36 => 157 | 37 => 163 | 38 => 167 | 39 => 173
+    | 40 => 179 | 41 => 181 | 42 => 191 | 43 => 193 | 44 => 197
+    | 45 => 199 | 46 => 211 | 47 => 223 | 48 => 227 | 49 => 229
+    | 50 => 233 | 51 => 239 | 52 => 241 | 53 => 251 | 54 => 257
+    | 55 => 263 | 56 => 269 | 57 => 271 | 58 => 277 | 59 => 281
+    | 60 => 283 | 61 => 293 | 62 => 307 | 63 => 311 | 64 => 313
+    | 65 => 317 | 66 => 331 | 67 => 337 | 68 => 347 | 69 => 349
+    | 70 => 353 | 71 => 359 | 72 => 367 | 73 => 373 | 74 => 379
+    | 75 => 383 | 76 => 389 | 77 => 397 | 78 => 401 | 79 => 409
```

Comment:
@Theodus thoughts on this vis-a-vis the style guide?

Reply:
Each branch on a separate line would be consistent with the rest of the stdlib. But a match of this size is unusual, so I don't have strong opinions about it.